ev3dev2/sensor/lego.py

```diff
@@ -626,16 +626,32 @@ def reset(self):
         self._ensure_mode(self.MODE_GYRO_ANG)
         self._direct = self.set_attr_raw(self._direct, 'direct', 17)
 
-    def wait_until_angle_changed_by(self, delta):
+    def wait_until_angle_changed_by(self, delta, honor_direction=False):
         """
         Wait until angle has changed by specified amount.
+
+        If ``honor_direction`` is True we will wait until angle has changed
+        by ``delta`` and with the correct sign.
+
+        If ``honor_direction`` is False (default) we will wait until angle has changed
+        by ``delta`` in either direction.
         """
         assert self.mode in (self.MODE_GYRO_G_A, self.MODE_GYRO_ANG,
                              self.MODE_TILT_ANG),\
             'Gyro mode should be MODE_GYRO_ANG, MODE_GYRO_G_A or MODE_TILT_ANG'
         start_angle = self.value(0)
-        while abs(start_angle - self.value(0)) < delta:
-            time.sleep(0.01)
+
+        if honor_direction:
+            if delta > 0:
```

Comment:
This feels like a lot of duplicated code. Can we combine this to be one loop? I think this condition might work:

```python
while (
        (abs(self.value(0) - start_angle) < delta)  # the sensor still hasn't moved far enough... 
        or (honor_direction and copysign(delta, 1) == copysign(self.value(0) - start_angle, 1))) # … or, it has moved in the wrong direction
```


I'm not entirely sure my line break structure will fly with the Python interpreter, but you get the gist.

Reply:
Personally I find the expanded/verbose model a little easier to follow :)

Reply:
I just want to remove the duplicated loop code... whether that's with a more clever loop condition or something else doesn't matter to me.